### ISSUE TITLE
Half days?

### ISSUE THREAD

**Issue by acp31** [airbnb/react-dates#123]
Is there a way to block off part of a day? For example If I wanted to show that only part of the day has been booked but the remaining part is still available.

**Comment by ljharb** [comment]
What's your use case? Typically date pickers I've seen deal in "days" or "nights" as opposed to partial days.

**Comment by acp31** [comment]
I am making a calendar for a client that rents out expensive property. The client has requested for the calendar to show when a property is not booked for an entire day they want the days to be shaded halfway if the day is only partially booked.

**Comment by ljharb** [comment]
Thanks for the explanation - so the client rents it out hourly, or in what time chunks?

**Comment by acp31** [comment]
The clients wants to rent out their properties in half day chunks. For example the client wants the ability to show a potential new customer that a property might not be available in the morning

**Comment by ljharb** [comment]
Gotcha, thanks for explaining.

Currently we only support "days" (which is also "nights" given that they're usually displayed the same).

Although we don't need a "half day" feature ourselves, this seems like something reasonable to add - as long as we can work out all the usability questions. We'd need to have half-day highlighting (vertical or horizontal? configurable?), and some way of cleanly and accessibly differentiating between which half of the day you'd selected, and making sure it's very clear what a half day selection means.

**Comment by acp31** [comment]
Hey Thanks for all of your help. I was poking around the airbnb website and noticed that some airbnb users have a set of dates blocked for the start date and a set blocked for end dates. This lets a user trying to make a reservation book a day that another client is checking out. Is there a way to do this with the current react-dates calendar?

**Comment by jontelm** [comment]
@acp31 Maybe you mean showing nights like this "half"? 

http://onefinestay.github.io/react-daterange-picker/

**Comment by theinventor** [comment]
> @acp31 Maybe you mean showing nights like this "half"?
> 
> http://onefinestay.github.io/react-daterange-picker/

Yes - I'd also love this, it's now a common feature for lodging. For example; 

Someone books friday night and saturday night, they will be checking out sunday morning. having 'half' of sunday shaded indicates that for everyone, otherwise that day appears "completely open" when it's not. Same for check-in day.

I'm new here, does this just need someone to try implementing or does it need an approval? 

Thanks all - LOVE this library!
